src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java

```diff
@@ -51,13 +53,14 @@ private CallbackValidations() {
 
     @NotNull
     static Validation<String, String> hasCaseReference(CaseDetails theCase) {
-        Object caseReference = null;
-        return theCase != null
-            && theCase.getData() != null
-            && (caseReference = theCase.getData().get(ATTACH_TO_CASE_REFERENCE)) != null
-            && (caseReference instanceof String)
-            ? valid((String) caseReference)
-            : internalError("no case reference found: %s", String.valueOf(caseReference));
+        return caseRefValidator.validate(theCase);
+    }
+
+    private static Optional<String> validRef(Object reference) {
+        return Optional.of(reference)
+            .filter(ref -> ref instanceof String)
+            .map(ref -> ((String) ref).replaceAll("[^0-9]", ""))
+            .filter(ref -> !ref.isEmpty());
```

Comment:
Although this implementation is acceptable and safe normally in java process I see something as follows:
```java
some method definition(String reference) {
    Assert.notEmpty(reference, "message");

    String ref = com.google.common.base.Strings.emptyOrNull(reference.replaceAll("[^0-9]", ""));

    return Optional.ofNull(ref);
}
```

Reply:
and there is not type checking either :/ ... since this is a union json element type it is quite possible to get a wrong object turning up in here at some point